src/Iter.mo

```diff
@@ -208,4 +209,12 @@ module {
     });
     List.reverse<A>(result);
   };
-}
+
+  /// Sorted iterator.  Will iterate over *all* elements to sort them, necessarily.
+  public func sort<A>(xs : Iter<A>, compare : (A, A) -> Order.Order) : Iter<A> {
+    let a = toArrayMut<A>(xs);
+    Array.sortInPlace<A>(a, compare);
```

Comment:
Since `Iter` is getting data in a streaming fashion, I wonder if feeding the elements into a RBTree/Trie is more efficient than array-based merge sort?

Reply:
The `Trie` is not an ordered structure, so not relevant.  The `RBTree` could sort the elements, but would require a lot of re-balancing in the process.  It's not a clear win.

But I do see the need to explore this more in the future, especially once we have the reference implementation (this one).

Reply:
Another idea is to use `Heap` and lazily pop the top element when asked. `Heapify` takes `O(n)` time instead of `O(nlogn)`. Would be an interesting experiment for cycle counting.

Reply:
Okay, I agree that's interesting.  Let's try that too and compare.